### PR TITLE
fix(cell): store tiles to shared memory with no bank-conflicts.

### PR DESCRIPTION
**Pull request by haruhi55**
This PR implements bank-conflict-free storage of tiles from registers to shared memory and from shared memory to global memory.

### CODE REVIEW

File: examples/cpp/01_gemm/util.hpp [hunk 1]
```diff
@@ -1,35 +1,59 @@
...
+    float elapsed = 0.;
+
+    if (timeit) {
+        for (int i = 0; i < 5; ++i) {
+            cublasHgemm(handle, CUBLAS_OP_T /* transb*/, CUBLAS_OP_N, kN, kM,
+                        kK, &alf, B, kK, A, kK, &bet, C, kN);
         }
+        cudaDeviceSynchronize();
+
+        int iters = 20;
```
Comment (KuangjuX) [inline examples/cpp/01_gemm/util.hpp:35]: Do we need to include `iters` in the parameter list?

File: include/cell/copy/global_to_shared.hpp [hunk 2]
```diff
@@ -229,17 +223,16 @@ struct SharedToGlobalStorerImpl<Shared_, Global_, kRowExec_, kColExec_,
     static constexpr int kColExec = kColExec_;
 
     // strides to iterate over each 16x16 `BaseTile` in the shared memory
-    static constexpr int kSrcCstride = BaseShape::kCols * Shared::kColStride;
-    static constexpr int kDstCstride = BaseShape::kCols * Global::kColStride;
-
-    static constexpr int kRstride = BaseShape::kRows;
+    // static constexpr int kSrcCstride = BaseShape::kCols * Shared::kColStride;
```
Comment (KuangjuX) [inline include/cell/copy/global_to_shared.hpp:226]: Does this comment still serve a purpose?

File: examples/cpp/01_gemm/util.hpp (hunk 1 above)
Comment (haruhi55) [reply examples/cpp/01_gemm/util.hpp]: fixed.

File: include/cell/copy/global_to_shared.hpp (hunk 2 above)
Comment (haruhi55) [reply include/cell/copy/global_to_shared.hpp]: deleted.

**Review by KuangjuX — APPROVED** [review]
LGTM!

**Pull request merged**